app/controllers/api/floating_ips_controller.rb

```diff
@@ -1,4 +1,39 @@
 module Api
   class FloatingIpsController < BaseController
+    def create_resource(_type, _id = nil, data = {})
+      ems = ExtManagementSystem.find(data['ems_id'])
+      raise BadRequestError, "Create floating ip not supported for Provider: #{ems.name}" unless ems.respond_to?(:create_floating_ip_queue)
```

Comment:
We should use `SupportsFeatureMixin` for this, e.g. `ems.supports?(:create_floating_ip)

Reply:
You can also raise `.unsupported_reason(:feature)` for each of these